src/modules/src/Eryph.Modules.Controller/DataServices/VirtualMachineDataService.cs

```diff
@@ -34,7 +34,7 @@ public async Task<Option<Catlet>> GetByVMId(Guid id)
 
         public async Task<Option<Catlet>> GetVM(Guid id)
         {
-            var res = await _repository.GetByIdAsync(id);
+            var res = await _repository.GetBySpecAsync(new CatletSpecs.GetByVMId(id));
```

Comment:
I think this should use `new CatletSpecs.GetById(id)`. Currently, `New-Catlet` fails in the `UpdateCatletSaga`.

Reply:
yep, should be GetById